src/rules/__tests__/prefer-importing-jest-globals.test.ts

```diff
@@ -0,0 +1,168 @@
+import { TSESLint } from '@typescript-eslint/utils';
+import dedent from 'dedent';
+import rule from '../prefer-importing-jest-globals';
+import { espreeParser } from './test-utils';
+
+const ruleTester = new TSESLint.RuleTester({
+  parser: espreeParser,
+  parserOptions: {
+    ecmaVersion: 2015,
+  },
+});
+
+ruleTester.run('prefer-importing-jest-globals', rule, {
+  valid: [
+    {
+      code: dedent`
+        // with import
+        import { test, expect } from '@jest/globals';
+        test('should pass', () => {
+            expect(true).toBeDefined();
+        });
+        `,
+      parserOptions: { sourceType: 'module' },
+    },
+    {
+      code: dedent`
+        // with require
+        const { test, expect } = require('@jest/globals');
+        test('should pass', () => {
+            expect(true).toBeDefined();
+        });
+        `,
+    },
+    {
+      code: dedent`
+        import { it as itChecks } from '@jest/globals';
+        itChecks("foo");
+      `,
+      parserOptions: { sourceType: 'module' },
+    },
+    {
+      code: dedent`
+        const { test } = require('@jest/globals');
+        test("foo");
+      `,
+    },
+  ],
+  invalid: [
+    {
+      code: dedent`
+        #!/usr/bin/env node
+        describe("suite", () => {
+          test("foo");
+          expect(true).toBeDefined();
+        })
+      `,
+      output: dedent`
+        #!/usr/bin/env node
+        const { describe, test, expect } = require('@jest/globals');
+        describe("suite", () => {
+          test("foo");
+          expect(true).toBeDefined();
+        })
+    `,
+      parserOptions: { sourceType: 'script' },
+      errors: [
+        { endColumn: 9, column: 1, messageId: 'preferImportingJestGlobal' },
+      ],
+    },
+    {
+      code: dedent`
+        // with comment above
+        describe("suite", () => {
+          test("foo");
+          expect(true).toBeDefined();
+        })
+      `,
+      output: dedent`
+        // with comment above
+        const { describe, test, expect } = require('@jest/globals');
```

Comment:
hmm, ideally this should be inserted _above_ the comment, no? I guess it's hard to make this generically correct

Reply:
It was [requested](https://github.com/jest-community/eslint-plugin-jest/pull/1490#discussion_r1463693631) to have a comment at the top of the file.

To have the `require` on top of the file I need to identify if the 1st node is a comment or shebang. Unfortunately, I have no solution to detect this kind of identifier.

I tried this solution but `firstToken.type` is evaluated as `Identifier` instead of the conditions listed below.

```js
const firstToken = sourceCode.getFirstToken(sourceCode.ast, { includeComments: true });

if (
  firstToken &&
  (firstToken.type === 'LineComment' || firstToken.type === 'BlockComment')
) {
  return fixer.insertTextBeforeRange(
    [0, 0],
    `${createFixerImports(isModule, jestFunctionsToImport)}\n`,
  );
}
```

Maybe you have a solution to identify the type of `firstToken`?